UI/ControlMappingScreen.cpp

```diff
@@ -349,30 +289,148 @@ bool KeyMappingNewKeyDialog::key(const KeyInput &key) {
+	float scale = CalculateTextScale(dc, availWidth);
+
+	dc.SetFontScale(scale, scale);
+
+	Bounds textBounds(bounds_.x + paddingX + textPadding_.left, bounds_.y, availWidth, bounds_.h);
+
+	dc.DrawTextRect(binding_ ? bindingText_.c_str() : text_.c_str(), textBounds, style.fgColor, ALIGN_VCENTER | FLAG_WRAP_TEXT);
+	dc.SetFontScale(1.0f, 1.0f);
+}
+
+void KeyMappingDialog::CreatePopupContents(UI::ViewGroup *parent) {
 	using namespace UI;
 
 	auto km = GetI18NCategory("KeyMapping");
+	auto mc = GetI18NCategory("MappableControls");
+	auto di = GetI18NCategory("Dialog");
+
+	ScrollView *scroll = new ScrollView(ORIENT_VERTICAL, new LinearLayoutParams(FILL_PARENT, 45, 1.0f));
+	LinearLayout *items = new LinearLayoutList(ORIENT_VERTICAL);
+	items->SetSpacing(5.0f);
+
+	static const std::string hint = "Hint: you may map the same key to different PSP button to be able to press multiple one with a single key!";
+	parent->Add(new TextView(km->T(hint), FLAG_DYNAMIC_ASCII | FLAG_WRAP_TEXT, true, new LinearLayoutParams(Margins(10,0))));
+
+	std::vector<KeyDef> mappings;
+	KeyMap::KeyFromPspButton(pspBtn_, &mappings, false);
+
+	for (size_t i = 0; i < mappings.size(); i++) {
+		std::string deviceName = GetDeviceName(mappings[i].deviceId);
+		std::string keyName = KeyMap::GetKeyOrAxisName(mappings[i].keyCode);
+
+		LinearLayout *row = items->Add(new LinearLayout(ORIENT_HORIZONTAL, new LinearLayoutParams(FILL_PARENT, WRAP_CONTENT)));
+		row->SetSpacing(1.0f);
+
+		BindingChoice *c = row->Add(new BindingChoice(deviceName + "." + keyName, km->T("Press a key"), new LinearLayoutParams(FILL_PARENT, 45, 1.0f)));
+		c->OnClick.Add([=](EventParams &e) {
+			this->selected_ = c; 
+			this->selectedIndex_ = i+1;
+			c->SetBinding(true);
+			return UI::EVENT_CONTINUE;
+		});
+
+		Choice *d = row->Add(new Choice(km->T("Delete"), new LinearLayoutParams(WRAP_CONTENT, 45)));
+		d->OnClick.Add([=](EventParams &e) {
+			KeyMap::g_controllerMap[pspBtn_].erase(KeyMap::g_controllerMap[pspBtn_].begin() + i);
+			KeyMap::g_controllerMapGeneration++;
+			RecreateViews();
+			return UI::EVENT_CONTINUE;
+		});
+	}
+
+	BindingChoice *n = items->Add(new BindingChoice(km->T("Add new key binding"), km->T("Press a key"), new LinearLayoutParams(FILL_PARENT, 45)));
+	n->OnClick.Add([=](EventParams &e) {
+		this->selected_ = n;
+		this->selectedIndex_ = 0;
+		n->SetBinding(true);
+		return UI::EVENT_CONTINUE;
+	});
+
+	if (g_Config.bMouseControl) {
+		BindingChoice *m = items->Add(new BindingChoice(km->T("Add new mouse binding"), km->T("Press ESC to cancel"), new LinearLayoutParams(FILL_PARENT, 45)));
+		m->OnClick.Add([=](EventParams &e) {
+			this->selected_ = m;
+			this->selectedIndex_ = 0;
+			m->SetBinding(true);
+			g_Config.bMapMouse = true;
+			return UI::EVENT_CONTINUE;
+		});
+	}
+
+	items->Add(new Spacer(10.0));
+	Choice *b = items->Add(new Choice(di->T("Back"), new LinearLayoutParams(FILL_PARENT, 45)));
+	b->OnClick.Handle<UIScreen>(this, &UIScreen::OnBack);
```

Comment:
I'd suggest OnCancel and naming it cancel.  DR_BACK should mainly be used for screens rather than pop ups.

-[Unknown]